packages/logs/src/domain/trackConsoleLogs.ts

```diff
@@ -0,0 +1,49 @@
+import type { ClocksState } from '@datadog/browser-core'
+import { instrumentMethodAndCallOriginal, clocksNow, formatConsoleParameters } from '@datadog/browser-core'
+import type { StatusType } from './logger'
+
+export const ApiName = {
+  log: 'log',
+  debug: 'debug',
+  info: 'info',
+  warn: 'warn',
+} as const
+
+export const LogStatus = {
+  log: 'info',
+  debug: 'debug',
+  info: 'info',
+  warn: 'warn',
+} as const
+
+export type ApiNameType = typeof ApiName[keyof typeof ApiName]
+
+export interface ConsoleLog {
+  message: string
+  status: StatusType
+  startClocks: ClocksState
+}
+
+export function trackConsoleLogs(apis: ApiNameType[], callback: (consoleLog: ConsoleLog) => void) {
+  const proxies = apis.map((api) => startConsoleProxy(api, callback))
+  return {
+    stop: () => {
+      proxies.forEach((proxy) => proxy.stop())
+    },
+  }
+}
+
+/* eslint-disable no-console */
+function startConsoleProxy(api: ApiNameType, callback: (consoleLog: ConsoleLog) => void) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏is `eslint-disable no-console` needed?

Reply:
No, thx